backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContext.ExamplesDomain.cs

```diff
@@ -0,0 +1,112 @@
+using ExampleApp.Examples.Domain.Booking;
+using ExampleApp.Examples.Domain.Employees;
+using ExampleApp.Examples.Domain.Projects;
+using LeanCode.AppRating.DataAccess;
+using LeanCode.DomainModels.EF;
+using LeanCode.DomainModels.Ids;
+using LeanCode.Firebase.FCM;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace ExampleApp.Examples.Services.DataAccess;
+
+public partial class ExamplesDbContext : IAppRatingStore<Guid>
+{
+    public DbSet<Employee> Employees => Set<Employee>();
+    public DbSet<Project> Projects => Set<Project>();
+
+    public DbSet<ServiceProvider> ServiceProviders => Set<ServiceProvider>();
+    public DbSet<CalendarDay> CalendarDays => Set<CalendarDay>();
+    public DbSet<Timeslot> Timeslots => Set<Timeslot>();
+
+    public DbSet<PushNotificationTokenEntity<Guid>> PushNotificationTokens => Set<PushNotificationTokenEntity<Guid>>();
+    public DbSet<AppRating<Guid>> AppRatings => Set<AppRating<Guid>>();
+
+    private void ConfigureExampleAppConventions(ModelConfigurationBuilder configurationBuilder)
+    {
+        ConfigureId<EmployeeId>(configurationBuilder);
+        ConfigureId<ProjectId>(configurationBuilder);
+        ConfigureId<AssignmentId>(configurationBuilder);
+        ConfigureId<ServiceProviderId>(configurationBuilder);
+        ConfigureId<CalendarDayId>(configurationBuilder);
+        ConfigureId<TimeslotId>(configurationBuilder);
+    }
+
+    private void OnExampleModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.ConfigurePushNotificationTokenEntity<Guid>(false);
+        modelBuilder.ConfigureAppRatingEntity<Guid>(SqlDbType.PostgreSql);
+
+        modelBuilder.Entity<Employee>(e =>
+        {
+            e.HasKey(t => t.Id);
+
+            e.IsOptimisticConcurrent(addRowVersion: false);
+            e.Property<uint>("xmin").IsRowVersion();
+        });
+
+        modelBuilder.Entity<Project>(e =>
+        {
+            e.HasKey(t => t.Id);
+
+            e.OwnsMany(
+                p => p.Assignments,
+                inner =>
+                {
+                    inner.WithOwner(a => a.ParentProject).HasForeignKey(a => a.ParentProjectId);
+
+                    inner.ToTable("Assignments");
+                }
+            );
+
+            e.IsOptimisticConcurrent(addRowVersion: false);
+            e.Property<uint>("xmin").IsRowVersion();
+        });
+
+        modelBuilder.Entity<ServiceProvider>(e =>
+        {
+            e.HasKey(t => t.Id);
+
+            e.OwnsOne(t => t.Location);
```

Comment:
```suggestion
            e.ComplexProperty(t => t.Location);
```

It's not an entity after all. Same for other value objects.

Reply:
TIL

Reply:
Beware of the [limitations](https://devblogs.microsoft.com/dotnet/announcing-ef8-rc1/#current-limitations) though, most importantly:
> Allow complex type properties to be null. ([Issue #31376](https://github.com/dotnet/efcore/issues/31376))